File: 2. Arrays/readability/readability.c

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int countLetters = count_letters(text);
    int countWords = count_words(text);
    int countSentences = count_sentences(text);
    float L = (float)countLetters / (float)countWords * 100;
    float S = (float)countSentences / (float)countWords * 100;
    int grade = round(0.0588 * L - 0.296 * S - 15.8);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}
int count_letters(string text)
{
    int countLetters = 0;
    char originalText[strlen(text)];
    strcpy(originalText, text);
    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {
        if ((int)(originalText[i]) >= 65 && (int)(originalText[i]) <= 122)
        {
            countLetters += 1;
        }
    }
    return countLetters;
}


int count_words(string text)
{
    int countWords = 1;
    char originalText[strlen(text)];
    strcpy(originalText, text);
    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {
        if (isspace(originalText[i]))
        {
            countWords++;
        }
    }
    return countWords;
}

int count_sentences(string text)
{
    int countSentences = 0;
    char originalText[strlen(text)];
    strcpy(originalText, text);
    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {
        if (originalText[i] == '.' || originalText[i] == '?' || originalText[i] == '!')
        {
            countSentences++;
        }
    }
    return countSentences;
}

File: 6. Python/readability/readability.py
from cs50 import get_string

Text = get_string("Text: ")

def count_letters():
    letters_num = 0
    for c in Text:
        if c.isalpha():
            letters_num += 1
    return letters_num

def count_words():
    words = Text.split()
    words_num = len(words)
    return words_num

def count_sentences():
    sentences_num = 0
    for c in Text:
        if c == '.' or c == '?' or c == '!':
            sentences_num += 1
    return sentences_num

L = count_letters() / count_words() * 100
S = count_sentences() / count_words() * 100
index = 0.0588 * L - 0.296 * S - 15.8

if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {round(index)}")
